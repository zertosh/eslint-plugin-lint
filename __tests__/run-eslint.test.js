'use strict';

const path = require('path');
const child_process = require('child_process');

function sanitize(str) {
  return String(str).split(__dirname).join('');
}

function yarnInstall(cwd) {
  return child_process.spawnSync('yarn', ['install'], {cwd, encoding: 'utf8'});
}

function runESLint(cwd) {
  return child_process.spawnSync(
    'node_modules/.bin/eslint',
    [
      'has-bar.js',
      'has-foo.js',
    ],
    {cwd, encoding: 'utf8'},
  );
}

test('works with eslint 3.x', () => {
  const cwd = path.join(__dirname, '__fixtures__/eslint-3');

  const yarnRet = yarnInstall(cwd);
  expect(yarnRet.status).toEqual(0);
  expect(yarnRet.error).toBeUndefined();

  const eslintRet = runESLint(cwd);
  expect(sanitize(eslintRet.stderr)).toMatchSnapshot();
  expect(sanitize(eslintRet.stdout)).toMatchSnapshot();
});

test('works with eslint 4.x', () => {
  const cwd = path.join(__dirname, '__fixtures__/eslint-4');

  const yarnRet = yarnInstall(cwd);
  expect(yarnRet.status).toEqual(0);
  expect(yarnRet.error).toBeUndefined();

  const eslintRet = runESLint(cwd);
  expect(sanitize(eslintRet.stderr)).toMatchSnapshot();
  expect(sanitize(eslintRet.stdout)).toMatchSnapshot();
});
