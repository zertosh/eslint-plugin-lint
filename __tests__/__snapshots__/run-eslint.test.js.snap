// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with eslint 3.x 1`] = `""`;

exports[`works with eslint 3.x 2`] = `
"
/__fixtures__/eslint-3/has-bar.js
  1:1  error  Has \\"BAR\\"  lint/no-bar

/__fixtures__/eslint-3/has-foo.js
  1:1  warning  Has \\"FOO\\"  lint/no-foo

✖ 2 problems (1 error, 1 warning)

"
`;

exports[`works with eslint 4.x 1`] = `""`;

exports[`works with eslint 4.x 2`] = `
"
/__fixtures__/eslint-4/has-bar.js
  1:1  error  Has \\"BAR\\"  lint/no-bar

/__fixtures__/eslint-4/has-foo.js
  1:1  warning  Has \\"FOO\\"  lint/no-foo

✖ 2 problems (1 error, 1 warning)

"
`;
